### ISSUE TITLE
[bitnami/cilium] Hubble Relay not starts because of ssl issue? wrong configuration?

### ISSUE THREAD

**Issue by zentavr** [bitnami/charts#30264]
### Name and Version

bitnami/cilium 1.2.5

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Installed the chart with custom helm values (below). Hubble-Relay and Hubble-UI cannot start because of configuration error. Logs of hubble relay:
```
time="2024-11-07T03:59:44Z" level=info msg="Starting gRPC health server..." addr=":4222" subsys=hubble-relay
time="2024-11-07T03:59:44Z" level=info msg="Starting gRPC server..." options="{peerTarget:cilium-hubble-peers.sys-cilium.svc.cluster.local:4244 dialTimeout:30000000000 retryTimeout:30000000000 listenAddress::4245 healthListenAddress::4222 metricsListenAddress: log:0xc0006870a0 serverTLSConfig:0xc00012d680 insecureServer:false clientTLSConfig:0xc00012d440 clusterName:default insecureClient:false observerOptions:[0x2218120 0x2218200] grpcMetrics:<nil> grpcUnaryInterceptors:[] grpcStreamInterceptors:[]}" subsys=hubble-relay
time="2024-11-07T03:59:52Z" level=debug msg="Server tls handshake" config=tls-server keypair-sn=a2fa35e9bb3f250f72f57db784f2e73 subsys=hubble-relay
time="2024-11-07T04:00:02Z" level=debug msg="Server tls handshake" config=tls-server keypair-sn=a2fa35e9bb3f250f72f57db784f2e73 subsys=hubble-relay
time="2024-11-07T04:00:12Z" level=debug msg="Server tls handshake" config=tls-server keypair-sn=a2fa35e9bb3f250f72f57db784f2e73 subsys=hubble-relay
time="2024-11-07T04:00:14Z" level=warning msg="Failed to create peer client for peers synchronization; will try again after the timeout has expired" error="context deadline exceeded" subsys=hubble-relay target="cilium-hubble-peers.sys-cilium.svc.cluster.local:4244"
time="2024-11-07T04:00:22Z" level=debug msg="Server tls handshake" config=tls-server keypair-sn=a2fa35e9bb3f250f72f57db784f2e73 subsys=hubble-relay
time="2024-11-07T04:00:32Z" level=debug msg="Server tls handshake" config=tls-server keypair-sn=a2fa35e9bb3f250f72f57db784f2e73 subsys=hubble-relay
time="2024-11-07T04:00:42Z" level=debug msg="Server tls handshake" config=tls-server keypair-sn=a2fa35e9bb3f250f72f57db784f2e73 subsys=hubble-relay
time="2024-11-07T04:00:52Z" level=debug msg="Server tls handshake" config=tls-server keypair-sn=a2fa35e9bb3f250f72f57db784f2e73 subsys=hubble-relay
time="2024-11-07T04:01:00Z" level=debug msg="Server tls handshake" config=tls-server keypair-sn=a2fa35e9bb3f250f72f57db784f2e73 subsys=hubble-relay
```

If I do the test of GRPC - it's OK:
```
I have no name!@cilium-hubble-relay-655b897f9f-96hpm:/opt/bitnami/hubble-relay$ grpc-health-probe -v -addr=cilium-hubble-peers.sys-cilium.svc.cluster.local:4244 -rpc-timeout=2s -tls -tls-ca-cert=/certs/client/ca.crt -tls-client-cert=/certs/client/tls.crt -tls-client-key=/certs/client/tls.key
parsed options:
> addr=cilium-hubble-peers.sys-cilium.svc.cluster.local:4244 conn_timeout=1s rpc_timeout=2s
> tls=true
  > no-verify=false 
  > ca-cert=/certs/client/ca.crt
  > client-cert=/certs/client/tls.crt
  > client-key=/certs/client/tls.key
  > server-name=
> alts=false
> spiffe=false
establishing connection
connection established (took 8.168682ms)
time elapsed: connect=8.168682ms rpc=677.245µs
status: SERVING
```

### Are you using any custom parameters or values?

```yaml
---
extraDeploy:
  - apiVersion: v1
    kind: Pod
    metadata:
      name: "cilium-curl"
    spec:
      containers:
      - name: curl
        image: 'curlimages/curl:8.6.0'
        command:
          - "/bin/sh"
          - "-ec"
          - "while :; do echo '.'; sleep 5; done"
        resources:
          limits:
            cpu: 500m
            memory: 512Mi
      restartPolicy: Never

# Seems like it breaks Hubble Relay
clusterName: default

azure:
  enabled: false

aws:
  enabled: false

gcp:
  enabled: false

agent:
  cniPlugin:
    install: true
    uninstall: false
  resources:
    requests:
      cpu: 250m
      memory: 256Mi
    limits:
      cpu: 375m
      memory: 384Mi
  tolerations:
    - operator: Exists
  containerSecurityContext:
    enabled: true
    ## Running with spc_t type (designed for privileged operations, check capabilities)
    ## since the container is not executed as a privileged container by default. This
    ## should prevent issues with SELinux policies.
    seLinuxOptions:
      level: 's0'
      type: 'spc_t'
    runAsUser: 0
    runAsGroup: 0
    runAsNonRoot: false
    readOnlyRootFilesystem: true
    # > Changed
    privileged: true
    # > Changed
    allowPrivilegeEscalation: true
    capabilities:
      add:
        - BPF
        - CHOWN
        - DAC_OVERRIDE
        - FOWNER
        - KILL
        - NET_ADMIN
        - NET_RAW
        - IPC_LOCK
        - PERFMON
        - SETGID
        - SETUID
        - SYS_ADMIN
        - SYS_MODULE
        - SYS_RESOURCE
      drop: [ "ALL" ]
    seccompProfile:
      type: "RuntimeDefault"

operator:
  replicaCount: 1
  resources:
    requests:
      cpu: 250m
      memory: 256Mi
    limits:
      cpu: 375m
      memory: 384Mi
  tolerations:
    - operator: Exists

envoy:
  useDaemonSet: false
  resources:
    requests:
      cpu: 250m
      memory: 256Mi
    limits:
      cpu: 375m
      memory: 384Mi

hubble:
  tls:
    enabled: true
  relay:
    image:
      debug: true
    enabled: true
    replicaCount: 1
    resources:
      requests:
        cpu: 100m
        memory: 128Mi
      limits:
        cpu: 150m
        memory: 192Mi
  ui:
    enabled: true
    replicaCount: 1
    service:
      #type: LoadBalancer
      type: ClusterIP
      annotations: {}
    ingress:
      enabled: false
      pathType: ImplementationSpecific
      hostname: hubble.local
      ingressClassName: ""
      path: /
      annotations: {}
      tls: false
      selfSigned: false
    frontend:
      resources:
        requests:
          cpu: 100m
          memory: 128Mi
        limits:
          cpu: 150m
          memory: 192Mi
    backend:
      resources:
        requests:
          cpu: 100m
          memory: 128Mi
        limits:
          cpu: 150m
          memory: 192Mi

etcd:
  enabled: true
  replicaCount: 3
  persistence:
    storageClass: "nfs-client"
  initialClusterState: "existing"

```

### What is the expected behavior?

Hubble Relay and Hubble UI work

### What do you see instead?

Hubble Relay and Hubble UI not work

### Additional information

1. Very likely we would like to have the template here as well, because the customer can change the value name:
    https://github.com/bitnami/charts/blob/0a49b81709b631dcd2e3f4c71d2c33deb52fdd91/bitnami/cilium/templates/hubble-relay/configmap.yaml#L13
    ...like here:
    https://github.com/bitnami/charts/blob/0a49b81709b631dcd2e3f4c71d2c33deb52fdd91/bitnami/cilium/templates/configmap.yaml#L21
2. Probably there is an error with the certificates for Hubble Relay deployment. Should we use the SAME certificates for `tls-relay-*` and `tls-hubble-*`?
   https://github.com/bitnami/charts/blob/0a49b81709b631dcd2e3f4c71d2c33deb52fdd91/bitnami/cilium/templates/hubble-relay/configmap.yaml#L29-L34

   at the vendors helm-chart they are different:
   https://github.com/cilium/cilium/blob/3a988cd59c83355891f4d61815baf260caf6bf1f/install/kubernetes/cilium/templates/_extensions.tpl#L20-L38
   and...
   https://github.com/cilium/cilium/blob/3a988cd59c83355891f4d61815baf260caf6bf1f/install/kubernetes/cilium/templates/hubble-relay/configmap.yaml#L36-L38

**Comment by zentavr** [comment]
So the issue occurs definitely when the value `hubble.tls.autoGenerated.engine` is `helm`. When deployed the cilium-agents, installed cert-manager and redeployed with `hubble.tls.autoGenerated.engine` = `cert-manager` the container had started.

**Comment by dgomezleon** [comment]
Hi @zentavr 


> ### Additional information
> 1. Very likely we would like to have the template here as well, because the customer can change the value name:
>    https://github.com/bitnami/charts/blob/0a49b81709b631dcd2e3f4c71d2c33deb52fdd91/bitnami/cilium/templates/hubble-relay/configmap.yaml#L13
>    
>    ...like here:
>    https://github.com/bitnami/charts/blob/0a49b81709b631dcd2e3f4c71d2c33deb52fdd91/bitnami/cilium/templates/configmap.yaml#L21
> 2. Probably there is an error with the certificates for Hubble Relay deployment. Should we use the SAME certificates for `tls-relay-*` and `tls-hubble-*`?
>    https://github.com/bitnami/charts/blob/0a49b81709b631dcd2e3f4c71d2c33deb52fdd91/bitnami/cilium/templates/hubble-relay/configmap.yaml#L29-L34
>    
>    at the vendors helm-chart they are different:
>    https://github.com/cilium/cilium/blob/3a988cd59c83355891f4d61815baf260caf6bf1f/install/kubernetes/cilium/templates/_extensions.tpl#L20-L38
>    and...
>    https://github.com/cilium/cilium/blob/3a988cd59c83355891f4d61815baf260caf6bf1f/install/kubernetes/cilium/templates/hubble-relay/configmap.yaml#L36-L38

Thanks for reporting it, I have created a PR to fix these typos.

Regarding the issue, I was not able to reproduce it on GKE. Note that we tested it setting `hubble.tls.autoGenerated.engine=helm`. You can find the values used for testing [here](https://github.com/bitnami/charts/blob/main/.vib/cilium/runtime-parameters.yaml). Could you give it a try in your environment? Note that you may need to disable `gcp.enabled` depending where your cluster is running.

**Comment by zentavr** [comment]
@dgomezleon we had spin up a cluster using kubeadm, v.1.31.2. The only situation I recall which could impact on it was the next:

during the very first run of the chart we enabled hubble.ui and run it. The invocation was successful but somewhere later we had noticed a warning that hubble UI requires hubble.relay. So we enabled hubble.relay then run the chart again. We saw two pods, for the ui and for the relay. Ui pod had waited for the relay, relay pod had that issue with the certificates.

could this situation impact on the certification creation process?
